### PR TITLE
Address reviewer comments on #27906

### PR DESCRIPTION
**Pull request by drfloob**
@markdroth

### CODE REVIEW

File: src/core/ext/transport/chttp2/server/chttp2_server.cc [hunk 1]
```diff
@@ -851,23 +851,27 @@ grpc_error_handle Chttp2ServerAddPort(Server* server, const char* addr,
                                       grpc_channel_args* args,
                                       Chttp2ServerArgsModifier args_modifier,
                                       int* port_num) {
-  if (strncmp(addr, "external:", 9) == 0) {
+  if (addr == nullptr) {
+    return GRPC_ERROR_CREATE_FROM_STATIC_STRING(
+        "Invalid address: addr cannot be a nullptr.");
+  }
+  if (strlen(addr) >= 9 && strncmp(addr, "external:", 9) == 0) {
```
Comment (markdroth) [inline src/core/ext/transport/chttp2/server/chttp2_server.cc:858]: I don't think this `strlen()` check is needed.  If `addr` is shorter, `strncmp()` will stop at its null terminator.

File: src/core/ext/transport/chttp2/server/chttp2_server.cc [hunk 2]
```diff
@@ -851,23 +851,27 @@ grpc_error_handle Chttp2ServerAddPort(Server* server, const char* addr,
...
   std::vector<grpc_error_handle> error_list;
-  std::string parsed_addr = URI::PercentDecode(absl::NullSafeStringView(addr));
+  std::string parsed_addr = URI::PercentDecode(addr);
   // Using lambda to avoid use of goto.
   grpc_error_handle error = [&]() {
     grpc_error_handle error = GRPC_ERROR_NONE;
     if (absl::StartsWith(parsed_addr, kUnixUriPrefix)) {
       error = grpc_resolve_unix_domain_address(
-          absl::StripPrefix(parsed_addr, kUnixUriPrefix), &resolved);
+          addr + sizeof(kUnixUriPrefix) - 1, &resolved);
```
Comment (markdroth) [inline src/core/ext/transport/chttp2/server/chttp2_server.cc:871]: These still need to look at `parsed_addr` instead of `addr`, right?

**Review by markdroth — COMMENTED** [review]
Just a couple of minor comments.

File: src/core/ext/transport/chttp2/server/chttp2_server.cc (hunk 2 above)
Comment (drfloob) [reply src/core/ext/transport/chttp2/server/chttp2_server.cc]: yes, good catch. thanks.

File: src/core/ext/transport/chttp2/server/chttp2_server.cc (hunk 1 above)
Comment (drfloob) [reply src/core/ext/transport/chttp2/server/chttp2_server.cc]: Removed

File: src/core/ext/transport/chttp2/server/chttp2_server.cc [hunk 3]
```diff
@@ -868,10 +868,10 @@ grpc_error_handle Chttp2ServerAddPort(Server* server, const char* addr,
     grpc_error_handle error = GRPC_ERROR_NONE;
     if (absl::StartsWith(parsed_addr, kUnixUriPrefix)) {
       error = grpc_resolve_unix_domain_address(
-          addr + sizeof(kUnixUriPrefix) - 1, &resolved);
+          &parsed_addr[sizeof(kUnixUriPrefix) - 1], &resolved);
```
Comment (markdroth) [inline src/core/ext/transport/chttp2/server/chttp2_server.cc:871]: I suggest writing this as `parsed_addr.data() + sizeof(kUnixUriPrefix) - 1`.

**Review by markdroth — APPROVED** [review]

File: src/core/ext/transport/chttp2/server/chttp2_server.cc (hunk 3 above)
Comment (drfloob) [reply src/core/ext/transport/chttp2/server/chttp2_server.cc:871]: Actually, both of these solutions break when `addr` contains embedded null. Passing a c-str is not a viable solution here.

Comment (drfloob) [reply src/core/ext/transport/chttp2/server/chttp2_server.cc]: `absl::ConsumePrefix` does the job more efficiently than the previous iteration, only checking `absl::StartsWith(...)` once.

**Review by markdroth — APPROVED** [review]
Looks great!

**Comment by drfloob** [comment]
Errors are a timeout, and somehow a successful test that's reported here as a failure (Bazel C/C++ Dbg MacOS: https://source.cloud.google.com/results/invocations/1def316e-5f0a-430f-a12e-36310fcf232b)

**Pull request merged**